module/zfs/zfs_ioctl.c

```diff
@@ -5843,6 +5903,9 @@ zfs_ioctl_init(void)
 	zfs_ioctl_register_dataset_nolog(ZFS_IOC_USERSPACE_UPGRADE,
 	    zfs_ioc_userspace_upgrade, zfs_secpolicy_userspace_upgrade,
 	    POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY);
+	zfs_ioctl_register_dataset_nolog(ZFS_IOC_USEROBJSPACE_UPGRADE,
+	    zfs_ioc_userobjspace_upgrade, zfs_secpolicy_userspace_upgrade,
+	    POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY);
```

Comment:
Yes, I think we can get rid of this.

However, regarding to your request of ```zfs upgrade```, @ahrens suggested the similar thing before that is to support file system feature, but my concern is that it's a little bit too much to implement it in this patch. We should do that in a separate task.
Right now, my work around solution is to do ```zfs set version=current <fs>``` that will trigger the upgrade command successfully. Just my 2 cents.